src/main/java/io/minio/MinioClient.java

```diff
@@ -382,26 +382,6 @@ public MinioClient(String endpoint, int port, String accessKey, String secretKey
       throw new InvalidEndpointException("(null)", "null endpoint");
     }
 
-    // for valid URL endpoint, port and secure are ignored
-    HttpUrl url = HttpUrl.parse(endpoint);
```

Comment:
hm, the problem is caused by some assumptions made in the different constructors. Seems not as easy:

 `
public MinioClient(String endpoint) throws InvalidEndpointException, InvalidPortException {
    this(endpoint, 0, null, null, true);
  }
`

Strange assumption, that this has to be a secure connection, if the endpoint is not interpreted. I try some more cleanup...